src/context/mod.rs

```diff
@@ -217,7 +256,177 @@ pub trait ContextTimers<'ctx>: private::SealedContext<'ctx> {
 
 impl<'ctx, CTX: private::SealedContext<'ctx>> ContextTimers<'ctx> for CTX {}
 
-/// Trait exposing Restate service to service communication functionalities.
+/// # Service Communication
+///
+/// A handler can call another handler and wait for the response (request-response), or it can send a message without waiting for the response.
+///
+/// ## Request-response calls
+///
+/// Request-response calls are requests where the client waits for the response.
+///
+/// You can do request-response calls to Services, Virtual Objects, and Workflows, in the following way:
+///
+/// ```rust,no_run
+/// # #[path = "../../examples/services/mod.rs"]
+/// # mod services;
+/// # use services::my_virtual_object::MyVirtualObjectClient;
+/// # use services::my_service::MyServiceClient;
+/// # use services::my_workflow::MyWorkflowClient;
+/// # use restate_sdk::prelude::*;
+/// #
+/// # async fn greet(ctx: Context<'_>, greeting: String) -> Result<(), HandlerError> {
+///     // To a Service:
+///     let service_response = ctx
+///         .service_client::<MyServiceClient>()
+///         .my_handler(String::from("Hi!"))
+///         .call()
+///         .await?;
+///
+///     // To a Virtual Object:
+///     let object_response = ctx
+///         .object_client::<MyVirtualObjectClient>("Mary")
+///         .my_handler(String::from("Hi!"))
+///         .call()
+///         .await?;
+///
+///     // To a Workflow:
+///     let workflow_result = ctx
+///         .workflow_client::<MyWorkflowClient>("my-workflow-id")
+///         .run(String::from("Hi!"))
+///         .call()
+///         .await?;
+///     ctx.workflow_client::<MyWorkflowClient>("my-workflow-id")
+///         .interact_with_workflow()
+///         .call()
+///         .await?;
+///  #    Ok(())
+///  # }
+/// ```
+///
+/// 1. **Create a service client**.
+///     - For Virtual Objects, you also need to supply the key of the Virtual Object you want to call, here `"Mary"`.
+///     - For Workflows, you need to supply a workflow ID that is unique per workflow execution.
+/// 2. **Specify the handler** you want to call and supply the request.
+/// 3. **Await** the call to retrieve the response.
+///
+/// **Tip: No need for manual retry logic**:
+/// These calls are proxied by Restate, and get logged in the journal.
+/// In case of failures, Restate takes care of retries.
+///
+/// **Note: Workflow retention time**:
+/// Once the `run` handler of the workflow has finished, the other handlers can still be called up to the retention time of the workflow, by default 24 hours.
+/// This can be configured via the [Admin API](https://docs.restate.dev/references/admin-api/#tag/service/operation/modify_service) per Workflow definition by setting `workflow_completion_retention`.
```

Comment:
This feels in the wrong place, this is a property of the callee, here we're talking about the caller. For this type of stuff, perhaps it makes sense to expand the macros virtual object and workflow to specify what are the properties of these two entities.